sw/airborne/modules/nav/nav_survey_rectangle_rotorcraft.c

```diff
@@ -161,9 +161,18 @@ void nav_survey_rectangle_rotorcraft_setup(uint8_t wp1, uint8_t wp2, float grid,
   }
 }
 
-
 bool nav_survey_rectangle_rotorcraft_run(uint8_t wp1, uint8_t wp2)
 {
+  return nav_survey_rectangle_rotorcraft_run_dynamic(wp1, wp2, -1);
+}
+
+bool nav_survey_rectangle_rotorcraft_run_dynamic(uint8_t wp1, uint8_t wp2, float grid)
+{
+  if (grid > 0) {
```

Comment:
May you use the ternary operator here? I think it is easier to read the code than to have two ifs and one else.